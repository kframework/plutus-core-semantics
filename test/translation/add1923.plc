(module Foo (import) (export () (add1923))
  (define add1923 [[(lam x (lam y (builtin addInteger x y))) 19] 23])
)
