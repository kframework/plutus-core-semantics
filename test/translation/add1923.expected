<T>
  <plutus>
    .
  </plutus>
  <totalFunctions>
    6
  </totalFunctions>
  <totalReadyForDispatch>
    6
  </totalReadyForDispatch>
  <dispatchedFunctionsLeft>
    0
  </dispatchedFunctionsLeft>
  <functions>
    FunctionCellSetItem ( <function>
      <name>
        @ alloc
      </name>
      <params>
        size  .VarList
      </params>
      <env>
        .Map
      </env>
      <k>
        % alloc_counter_ptr = 0  % alloced = load % alloc_counter_ptr  %ret = % alloced  % alloced = add %size , % alloced  store % alloced , % alloc_counter_ptr  ret %ret , .Operands  .Instructions
      </k>
      <jumpId>
        -1
      </jumpId>
      <hasDispatchEntry>
        true
      </hasDispatchEntry>
      <readyForDispatch>
        true
      </readyForDispatch>
    </function> ) FunctionCellSetItem ( <function>
      <name>
        @ dispatch
      </name>
      <params>
        closure  param  .VarList
      </params>
      <env>
        .Map
      </env>
      <k>
        % fptr = load %closure  % cond = cmp eq % fptr , 4  br % cond , label4  % cond = cmp eq % fptr , 5  br % cond , label5  #block ( label5 )  call @fresh5 ( %closure , %param , .Operands )  #block ( label4 )  call @fresh4 ( %closure , %param , .Operands )  .Instructions
      </k>
      <jumpId>
        -1
      </jumpId>
      <hasDispatchEntry>
        true
      </hasDispatchEntry>
      <readyForDispatch>
        true
      </readyForDispatch>
    </function> ) FunctionCellSetItem ( <function>
      <name>
        @ init
      </name>
      <params>
        .VarList
      </params>
      <env>
        .Map
      </env>
      <k>
        % alloc_counter_ptr = 0  store 1 , % alloc_counter_ptr  .Instructions
      </k>
      <jumpId>
        -1
      </jumpId>
      <hasDispatchEntry>
        true
      </hasDispatchEntry>
      <readyForDispatch>
        true
      </readyForDispatch>
    </function> ) FunctionCellSetItem ( <function>
      <name>
        @add1923
      </name>
      <params>
        .VarList
      </params>
      <env>
        .Map
      </env>
      <k>
        %fresh3 , .LValues = call @ alloc ( 1 , .Operands )  store 4 , %fresh3  %fresh1 , .LValues = call @ dispatch ( %fresh3 , 19 , .Operands )  %ret , .LValues = call @ dispatch ( %fresh1 , 23 , .Operands )  ret %ret , .Operands  .Instructions
      </k>
      <jumpId>
        -1
      </jumpId>
      <hasDispatchEntry>
        true
      </hasDispatchEntry>
      <readyForDispatch>
        true
      </readyForDispatch>
    </function> ) FunctionCellSetItem ( <function>
      <name>
        @fresh4
      </name>
      <params>
        x  .VarList
      </params>
      <env>
        .Map
      </env>
      <k>
        %ret , .LValues = call @ alloc ( 2 , .Operands )  store 5 , %ret  %fresh7 = add %ret , 1  store %x , %fresh7  ret %ret , .Operands  .Instructions
      </k>
      <jumpId>
        4
      </jumpId>
      <hasDispatchEntry>
        true
      </hasDispatchEntry>
      <readyForDispatch>
        true
      </readyForDispatch>
    </function> ) FunctionCellSetItem ( <function>
      <name>
        @fresh5
      </name>
      <params>
        y  .VarList
      </params>
      <env>
        0 |-> x
      </env>
      <k>
        % addr = add %closure , 1  %fresh8 = load % addr  %ret = add %fresh8 , %y  ret %ret , .Operands  .Instructions
      </k>
      <jumpId>
        5
      </jumpId>
      <hasDispatchEntry>
        true
      </hasDispatchEntry>
      <readyForDispatch>
        true
      </readyForDispatch>
    </function> )
  </functions>
</T>
